src/transaction/script.c

```diff
@@ -0,0 +1,228 @@
+#include <stdbool.h>  // bool
+#include <string.h>   // memset, explicit_bzero, memmove
+
+#include "script.h"
+
+#include "../constants.h"
+#include "opcodes.h"
+
+#include "../globals.h"
+
+
+/**
+ * Identifies if the given script is P2PKH
+ */
+bool identify_p2pkh_script(buffer_t *in, size_t script_len) {
+    uint8_t p2pkh[] = {OP_DUP, OP_HASH160, PUBKEY_HASH_LEN, OP_EQUALVERIFY, OP_CHECKSIG};
+    size_t offset = 0;
+
+    if ((in == NULL) || ((script_len != 25) && (script_len != 31)) || (in->size - in->offset < script_len)) {
+        PRINTF("[-] [Identify P2PKH script] invalid args\n");
+        return false;
+    }
+
+    if (script_len == 31) {
+      // check timelock
+      if (in->ptr[in->offset] != 4) {
+          return false;
+      }
+      // skip push_4 (1 byte) + timestamp (4 bytes)
+      offset += 5;
+      if (in->ptr[in->offset + offset] != OP_GREATERTHAN_TIMESTAMP) {
+          return false;
+      }
+      // skip OP_GREATERTHAN_TIMESTAMP (1 byte)
+      offset += 1;
+    }
```

Comment:
We allow identifying and parsing outputs with timelock but they are not supported yet.